xcm/src/v1/multilocation.rs

```diff
@@ -323,6 +323,60 @@ impl MultiLocation {
 		}
 		Ok(())
 	}
+
+	/// Mutate `self` so that it represents the same location from the point of view of `target`.
+	/// The context of `self` is provided as `ancestry`.
+	///
+	/// Does not modify `self` in case of overflow.
+	pub fn reanchor(&mut self, target: &MultiLocation, ancestry: &MultiLocation) -> Result<(), ()> {
+		// TODO: Optimize this.
```

Comment:
So I realized that what we're essentially doing here is to first find the common ancestor between `self` and `target`, then append the necessary junctions that will allow us to reach `self` from the common ancestor, and then we'd simplify the result, given `ancestry`.

I suspect what we can do here is to write the common ancestor from the perspective of `target`, that way we don't have to simplify at the end of the algorithm.

Reply:
Right. Hence the TODO :)